manta-pay/src/crypto/ecc/arkworks.rs

```diff
@@ -516,3 +517,82 @@ where
         )
     }
 }
+
+impl<C, CV> FixedBaseScalarMul<Compiler<C>> for GroupVar<C, CV>
+where
+    C: ProjectiveCurve,
+    CV: CurveVar<C, ConstraintField<C>>,
+{
+    type Base = C;
+
+    fn fixed_base_scalar_mul<I, T>(
+        precomputed_bases: I,
+        scalar: &Self::Scalar,
+        compiler: &mut Compiler<C>,
+    ) -> Self
+    where
+        I: IntoIterator<Item = T>,
+        T: Borrow<Self::Base>,
+    {
+        let _ = compiler;
+        let mut result = CV::zero();
+        let scalar_bits = scalar
+            .0
+            .to_bits_le()
+            .expect("Bit decomposition is not allowed to fail.");
+        for (bit, base) in scalar_bits.into_iter().zip(precomputed_bases.into_iter()) {
+            result = bit
+                .select(&(result.clone() + *base.borrow()), &result)
+                .expect("Conditional select is not allowed to fail. ");
+        }
+        Self::new(result)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::config::Bls12_381_Edwards;
+    use manta_crypto::{
+        algebra::Group, arkworks::r1cs_std::groups::curves::twisted_edwards::AffineVar,
+        constraint::measure::Measure, eclair::bool::AssertEq, rand::OsRng,
+    };
+
+    /// Constraint Field Type
+    type ConstraintField<C> = <<C as ProjectiveCurve>::BaseField as Field>::BasePrimeField;
+
+    /// Compiler Type
+    type Compiler<C> = R1CS<ConstraintField<C>>;
+
+    /// Scalar Field Element
+    pub type Scalar<C> = Fp<<C as ProjectiveCurve>::ScalarField>;
+
+    #[test]
+    fn fixed_base_mul() {
+        let mut cs = Compiler::<Bls12_381_Edwards>::for_proofs();
+        let scalar = Scalar::<Bls12_381_Edwards>::gen(&mut OsRng);
+        let base = Bls12_381_Edwards::prime_subgroup_generator();
+        let precomputed_table =
+            core::iter::successors(Some(base), |base| Some(*base + *base)).take(256);
```

Comment:
Can we move this in `manta-crypto` so that it's available to the circuit-generation code?

Reply:
Moved to `manta-crypto::arkworks::algebra`